omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_0/EncounterSearchHandler2_0.java

```diff
@@ -69,13 +69,24 @@ public PageableResult search(RequestContext context) throws ResponseException {
 				Patient.class)).getByUniqueId(patientUuid);
 		EncounterType encounterType = ((EncounterTypeResource1_8) Context.getService(RestService.class)
 				.getResourceBySupportedClass(EncounterType.class)).getByUniqueId(encounterTypeUuid);
+
+		Collection<Visit> visits = new ArrayList<>();
+		if(visitsUuid.length>0) {
+
+			for (String uuid : visitsUuid) {
+				Visit visit = ((VisitResource1_9) Context.getService(RestService.class).getResourceBySupportedClass(Visit.class)).getByUniqueId(uuid);
+				visits.add(visit);
+			}
+		}
```

Comment:
Might as well wait to do all this once you are actually doing the query below, rather than pre-emptively, since patient is current required.  If we want to make patient no longer required, and query on visit only, then we need to make sure the patient null check does not wrap everything.  See suggested code below.

Reply:
Okay.I'll add up those validations for visits before the query itself.